clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -796,11 +821,26 @@ protected void startConsumingAsLeader(PartitionConsumptionState partitionConsump
     }
     final PubSubTopic leaderTopic = offsetRecord.getLeaderTopic(pubSubTopicRepository);
     final PubSubTopicPartition leaderTopicPartition = partitionConsumptionState.getSourceTopicPartition(leaderTopic);
-    final long leaderStartOffset = partitionConsumptionState
+    long leaderStartOffset = partitionConsumptionState
         .getLeaderOffset(OffsetRecord.NON_AA_REPLICATION_UPSTREAM_OFFSET_MAP_KEY, pubSubTopicRepository);
+    String leaderSourceKafkaURL = leaderSourceKafkaURLs.iterator().next();
+    if (leaderStartOffset < 0 && leaderTopic.isRealTime()) {
+      // TODO: Add description.
+      TopicSwitch topicSwitch = partitionConsumptionState.getTopicSwitch().getTopicSwitch();
+      if (topicSwitch == null) {
+        throw new VeniceException(
+            "New leader does not have topic switch, unable to switch to realtime leader topic: "
+                + leaderTopicPartition);
+      }
+      leaderStartOffset = calculateLeaderUpstreamOffsetWithTopicSwitch(
```

Comment:
Shall we update PCS with this new number?

Reply:
I think it is done in the below line 835. 
Also, (just for my understanding) if it is not done here, it will be done after next value being processed as well, is that right?

Reply:
I missed that line. PCS is updated.